Add docs for monorepo (#2107)
vercel/docs: pages/docs/git-integrations/index.mdx

/>
<Caption>Configuring the Production Branch of your Project.</Caption>

## Monorepos

If you'd like to deploy multiple different directories within the same Git repository,

> you can do so by creating a separate Project for each directory and configuring the [Root Directory setting](/docs/build-step?query=root%20directory#root-directory) for it.

Reviewer: ```suggestion
you can do so by creating a separate [Project](/docs/platform/projects) for each directory and configuring the [Root Directory setting](/docs/build-step?query=root%20directory#root-directory) for it.
```

The easiest way to achieve this is by [importing your Git repository](/docs/v2/git-integrations#quick-installation) multiple times and selecting a different
Root Directory every time – until one Project was created for each of the relevant directories.

<Image
  src={`${process.env.ASSETS}/docs/common/monorepo-import.png`}
  width={1284 / 2}
  height={966 / 2}
  shadow
  borderRadius
/>
<Caption>Selecting a Root Directory for one of your new Projects.</Caption>

Once you've created a separate Project for each of the directories within your Git repository,
every commit will issue a Deployment for all connected Projects and display the resulting URLs on your pull requests and commits:

<Image
  src={`${process.env.ASSETS}/docs/common/monorepo-pr.png`}
  width={1809 / 2}
  height={920 / 2}
/>
<Caption>An example of Deployment URLs provided by a Git Integration.</Caption>

> Up to 20 Projects can be connected to the same Git repository. If you need to connect more, please [contact us](/support/request).

Reviewer 2: ```suggestion
Up to 20 Projects can be connected to the same Git repository. If you need to connect more, please [contact our Sales Team](/contact/sales).
```

> Up to 20 Projects can be connected to the same Git repository. If you need to connect more, please [contact our Sales Team](/contact/sales).

Reviewer: Can we add this to the Limits page also?

Author: Under the list of [General Limits](https://vercel.com/docs/platform/limits#general-limits) or as a standalone section?

Reviewer: The standalone section would be great @shuding - very similar to the 'Domains' heading.

Author: @mcsdevv Just had a chat with @leo and we are adding a "Git Repository Connections" section to that page :+1:

> How many Vercel Projects can be connected with the same Git repository is [decided by your pricing plan](/docs/platform/limits#general-limits).

Reviewer 2: ```suggestion
How many Vercel Projects can be connected with the same Git repository is decided by your plan. Learn more about the limits [here](/docs/platform/limits#general-limits).
```

### Using Monorepos with Vercel CLI

> Because it's only possible to link a single Vercel Project when running the vercel command, we recommend cloning your Git repository multiple times and then running vercel for each one to deploy them.

Reviewer 2: ```suggestion
Because it's only possible to link a single Vercel Project when running the `vercel` command, we recommend cloning your Git repository multiple times and then running `vercel` for each one to deploy them.
```

> The same applies to the vercel dev command (local Development).

Reviewer 2: ```suggestion
The same applies to the `vercel dev` command (local Development).
```
